Sources/MapboxMaps/Foundation/ResourceOptionsManager.swift

```diff
@@ -0,0 +1,147 @@
+import Foundation
+@_implementationOnly import MapboxCommon_Private
+
+/// Convenience class that manages a global `ResourceOptions`
+///
+/// It's possible to create `ResourceOptionsManager` instances as you need them,
+/// however it's convenient to use the default object (`default`).
+///
+/// For example, we recommend that the Mapbox access token be set in
+/// `application(_:didFinishLaunchingWithOptions:)` rather than relying on the
+/// value in your application's Info.plist:
+///
+///     ```
+///     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+///         // Override point for customization after application launch.
+///         ResourceOptionsManager.default.resourceOptions.accessToken = "overridden-access-token"
+///         return true
+///     }
+///     ```
+public class ResourceOptionsManager {
+
+    /// Default instance
+    ///
+    /// This shared instance is used by the default initializers
+    /// for `ResourceOptions` and `MapInitOptions`.
+    ///
+    /// The application's Info.plist will be searched for a valid access token
+    /// under the key `MBXAccessToken`
+    ///
+    /// A valid access token must be provided or found.
+    public static var `default`: ResourceOptionsManager {
+        if defaultInstance == nil {
+            defaultInstance = ResourceOptionsManager(accessToken: nil)
```

Comment:
on iOS, do you throw exceptions when the accessToken is null?

Reply:
Not at the moment, does Android? Also, `nil` in this context means "search for the token".